docs/jumpstart/week-9.md

```diff
@@ -0,0 +1,343 @@
+
+With this foreign key in place, we can now make an order for any product, and keep order information separate from product info. But we'd like to take this one step further: a single order should be able to contain multiple products. As we briefly mentioned earlier, this is impossible with the current, two-table setup, because our `orders` table would need to have an arbitrary number of `product_id`s. The solution is to create another table called a <strong>join</strong> table. It has one purpose: to <em>join</em>, or connect, two other tables, so that we can establish what we call a <em>many-to-many</em> relationship -- put simply, this means that an order can contain an arbitrary many products.
+
+Our new table only needs two columns: an order id and a product id. Every entry in this table represents something like a single item in a shopping basket. That item belongs to a particular basket (the order), and points to a product. So, we might create a table like this:
+```sql
+CREATE TABLE order_items (
+    product_id integer REFERENCES products,
+    order_id integer REFERENCES orders
+);
+```
+This is an example of a table that needs no primary key. We might add one if we want, but it may not be useful because the only purpose this table serves is to join two existing tables. If we need to refer to a specific entry in this table, we can use the combination of `order_id` and `product_id`, since no two rows should have the same value for both. We'll see how to do this in the next section.
+
+## Other Constraints
+As we mentioned earlier, `PRIMARY KEY` is technically just a combination of the `UNIQUE` and `NOT NULL` constraints. These constraints are self-explanatory; `UNIQUE` requires that a column's value is unique with respect to all other rows in the table, and `NOT NULL` requires that a column's value is not null (like in Javascript, null just represents the absence of a value). In general, most columns should be `NOT NULL`. Only allow columns be `NULL` if it makes sense for there to be no data.
+
+Note: when we say "require" in terms of constraints, we mean that any attempt to create or update an entry in the table that will violate the constraint will result in an error. This enables us to assume that all entries in a table satisfy the table's constraints.
+
+Here is an example of how we might define these constraints:
+```sql
+CREATE TABLE products (
+    id serial UNIQUE NOT NULL,
+    name text NOT NULL,
+    price numeric
+);
+```
+What if we no columns in a table are unique by themselves, but the combination of two or more is unique? For example, recall `order_items`, our join table from earlier:
```

Comment:
The beginning of this sentence is a bit confusing. I think that the "we" should not be there